src/note/nullifier.rs

```diff
@@ -30,6 +31,16 @@ impl Nullifier {
         Nullifier(extract_p(&pallas::Point::random(rng)))
     }
 
+    /// Deserialize the nullifier from a byte array.
+    pub fn from_bytes(bytes: &[u8; 32]) -> CtOption<Self> {
+        pallas::Base::from_bytes(bytes).map(Nullifier)
```

Comment:
Hmm, do we want to check that the nullifier is a valid x-coordinate (rather than simply a valid Fp element)? The spec types nullifiers as `Fp`, so this is technically fine as-is, but we know a nullifier can't ever be an invalid x-coordinate. cc @daira

Reply:
nf is intentionally typed as an integer in [0, q_P). The circuit checks that this is an output of DeriveNullifier (and hence Extract_P), but there is no such rule outside the circuit. Checking it outside the circuit would require taking a square root / Jacobi symbol, which is expensive.